src/components/state-history-chart-line.js

```diff
@@ -155,14 +155,14 @@ class StateHistoryChartLine extends LocalizeMixin(PolymerElement) {
         domain === "climate" ||
         domain === "water_heater"
       ) {
-        const isHeating =
-          domain === "climate"
-            ? (state) => state.attributes.hvac_action === "heating"
-            : (state) => state.state === "heat";
-        const isCooling =
-          domain === "climate"
-            ? (state) => state.attributes.hvac_action === "cooling"
-            : (state) => state.state === "cool";
+        const isHeating = (state) =>
```

Comment:
Now we're no longer consistent but could change on a per-state basis. I think that we should go through all the states and see if any has an hvac action. We should calculate similarly to how we calculate `hasTargetRange`

Reply:
I didn't know the presence of `hvac_action` can change over time, but the code now supports it. It's only accounted for within the given timeframe, but I think that's as good as it gets, unless the platform has some way of signaling that `hvac_action` might be present.